fix: improve perf for sortByMarketCap

KVNLS opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already.
Disclaimer: Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### 📝 Description

Today we have some performance issues when we open the Swap page, because sortByMarkCap can be slow.
This PR improves the performance of the function.

### ❓ Context

- **Impacted projects**: `llm`, `lld` <!-- Add the list of end user projects impacted by the change inside of the ` `, like so `LLM, LLD`. -->
- **Linked resource(s)**: [] <!-- Attach any ticket number if relevant inside the brackets, like so [LIVE-0000]. (JIRA / Github issue number) -->
- https://ledgerhq.atlassian.net/browse/LIVE-9663
- https://ledgerhq.atlassian.net/browse/LIVE-9997

### ✅ Checklist

- [x] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
- [x] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
- [x] **No breaking changes** <!-- If there are breaking changes, please explain why. -->

### 📸 Demo

<!--
For visual features, please attach screenshots or video recordings to demonstrate the changes.
For libraries, you can add a code sample.
For bugfixes, you can drop this section.
-->



Before
<img width="794" alt="Screenshot 2023-10-30 at 17 14 04" src="https://github.com/LedgerHQ/ledger-live/assets/31533861/5300da6a-a856-4885-95ec-4e539a3b7967">

After
<img width="1302" alt="Screenshot 2023-10-31 at 14 59 17" src="https://github.com/LedgerHQ/ledger-live/assets/31533861/fcbed26d-f4fa-4f87-97ba-1bfa7da8acad">

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

<!-- If any of the expectations are not met please explain the reason in detail. -->

gre commented on libs/ledger-live-common/src/currencies/sortByMarketcap.ts:
```diff
@@ -31,23 +31,32 @@ function lazyLoadTickerMap() {
-        list.splice(i, 1);
-        prependList.push(item);
-      }
+  currencies.forEach(currency => {
```
could you also use `for of` syntax here so we have similar code style as the previous for loop?

alternatively it seems the logic here is also fundamentally a
```
const remainingCurrencies = currencies.filter(c => !sortedCurrenciesMap.has(getCurrencyId(c)));
```
which may be similar in perf but simpler code ? 🤔

KVNLS commented on libs/ledger-live-common/.unimportedrc.json:
```diff
@@ -303,7 +303,8 @@
     "src/index.ts",
     "src/jest.d.ts",
     "src/postOnboarding/mock.ts",
-    "src/reactNativeSvg.d.ts"
```
@valpinkman we need conventions on our file structure otherwise it will be hard to maintain. Something like __tests__/mock.ts in that case we will be able to exclude __tests__ pattern.

gre replied:
we could also consider currencies/mock to be a valid entry point btw, because it is somehow (for the tests).
the problem rely more on the fact live-common is a multi entry point library.

KVNLS commented:
Update with the latest version:
<img width="788" alt="image" src="https://github.com/LedgerHQ/ledger-live/assets/31533861/e885711c-d968-432d-9cd7-bba24a040c3d">
@gre I added a unit test covered by a snapshot. My first iteration did not cover it so I updated the algo.
